src/Workspaces/Core/Portable/Workspace/Solution/RecoverableTextAndVersion.cs

```diff
@@ -15,129 +16,117 @@ namespace Microsoft.CodeAnalysis
     /// <summary>
     /// A recoverable TextAndVersion source that saves its text to temporary storage.
     /// </summary>
-    internal class RecoverableTextAndVersion : ValueSource<TextAndVersion>, ITextVersionable
+    internal sealed class RecoverableTextAndVersion : ValueSource<TextAndVersion>, ITextVersionable
     {
-        private readonly ITemporaryStorageServiceInternal _storageService;
+        private readonly HostWorkspaceServices _services;
```

Comment:
possible to relax to SolutionServices?

Reply:
Not easily. Looking into this now.

Reply:
i think you only use this for GetRequiredService below.  so this should be relaxable.  nbd though.,

Reply:
You are right. Done. I was thinking about the caller where the Workspace is still needed.